trading_calendars/exchange_calendar_xnze.py

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time
+from pandas.tseries.holiday import (
+    DateOffset,
+    EasterMonday,
+    GoodFriday,
+    Holiday,
+    MO,
+    previous_workday,
+    weekend_to_monday,
+)
+from pytz import timezone
+
+from .common_holidays import (
+    new_years_day,
+    christmas,
+    weekend_christmas,
+    boxing_day,
+    weekend_boxing_day,
+)
+
+from .trading_calendar import (
+    TradingCalendar,
+    HolidayCalendar
+)
+
+# Regular Holidays
+# ----------------
+NewYearsDay = new_years_day()
+
+DayAfterNewYearsDay = Holiday(
+    "Day after New Year's Day",
+    month=1,
+    day=2,
+)
+
+WaitangiDay = Holiday(
+    "Waitangi Day",
+    month=2,
+    day=6,
+    observance=weekend_to_monday,
+)
+
+AnzacDay = Holiday(
+    "Anzac Day",
+    month=4,
+    day=25,
+    observance=weekend_to_monday,
+)
+
+QueensBirthday = Holiday(
+    "Queen's Birthday",
+    month=6,
+    day=1,
+    offset=DateOffset(weekday=MO(1)),
+)
+
+LabourDay = Holiday(
+    "Labour Day",
+    month=10,
+    day=1,
+    offset=DateOffset(weekday=MO(4)),
+)
+
+Christmas = christmas()
+
+WeekendChristmas = weekend_christmas()
+
+BoxingDay = boxing_day()
+
+WeekendBoxingDay = weekend_boxing_day()
+
+
+# Early Closes
+# ------------
+BusinessDayPriorToChristmasDay = Holiday(
+    "Business Day prior to Christmas Day",
+    month=12,
+    day=25,
```

Comment:
I think this should be `24`. This is basically like saying Christmas Eve is an early close unless it's a weekend, in which case go back to the previous workday.

Reply:
Talked to Andrew in person, what he has actually works. I was thinking of `previous_friday`, which is confusingly similar:

```
> previous_workday?
Signature: previous_workday(dt)
Docstring: returns previous weekday used for observances

> previous_friday?
Signature: previous_friday(dt)
Docstring: If holiday falls on Saturday or Sunday, use previous Friday instead.
```